portals/admin/source/src/app/components/AdvancedSettings/TenantConfSave.jsx

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, {
+    useReducer, useState, Suspense, lazy, useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useIntl, FormattedMessage } from 'react-intl';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+    Box, Grid, Button,
+} from '@material-ui/core';
+import { green } from '@material-ui/core/colors';
+import API from 'AppData/api';
+import Alert from 'AppComponents/Shared/Alert';
+import { Progress } from 'AppComponents/Shared';
+import ContentBase from 'AppComponents/AdminPages/Addons/ContentBase';
+
+const MonacoEditor = lazy(() => import('react-monaco-editor' /* webpackChunkName: "TeantConfAddMonacoEditor" */));
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        marginBottom: theme.spacing(15),
+    },
+    error: {
+        color: theme.palette.error.dark,
+    },
+    dialog: {
+        minWidth: theme.spacing(150),
+
+    },
+    tenantConfHeading: {
+        marginBottom: theme.spacing(1),
+    },
+    showSampleButton: {
+        marginTop: theme.spacing(2),
+    },
+    helperText: {
+        color: green[600],
+        fontSize: theme.spacing(1.6),
+        marginLeft: theme.spacing(1),
+    },
+    infoBox: {
+        marginTop: theme.spacing(1),
+        marginBottom: theme.spacing(2),
+    },
+    buttonBox: {
+        marginTop: theme.spacing(2),
+        marginBottom: theme.spacing(2),
+        marginLeft: theme.spacing(1),
+    },
+    saveButton: {
+        marginRight: theme.spacing(2),
+    },
+}));
+
+/**
+ * Reducer
+ * @param {JSON} state The second number.
+ * @returns {Promise}
+ */
+function reducer(state, { field, value }) {
+    switch (field) {
+        case 'tenantConf':
+            return { ...state, [field]: value };
+        case 'tenantConfSchema':
+            return { ...state, [field]: value };
+        case 'editDetails':
+            return value;
+        default:
+            return state;
+    }
+}
+
+/**
+ * Render a list
+ * @returns {JSX} Header AppBar components.
+ */
+function TenantConfSave(props) {
+    const classes = useStyles();
+    const {
+        updateList, history,
```

Comment:
shall we use `useHistory` from react-router, it seems like parent component is passing it's history object to this component

Reply:
fixed